gating/check/pre

```diff
@@ -24,3 +26,27 @@ fi
 # yaml library are present for ansible to work.
 source "$(readlink -f $(dirname ${0}))/../gating_prerequisites.sh"
 
+# Install pre-requisites for the different types of tests
+distro_pkgs_to_install=""
+py_pkgs_to_install=""
+
+if [[ $RE_JOB_IMAGE =~ .*mnaio.* ]]; then
+  dpkg-query --list | grep curl &>/dev/null || distro_pkgs_to_install+="curl "
+fi
+
+if [[ $RE_JOB_ACTION == "tox-test" ]]; then
+  dpkg-query --list | grep wget &>/dev/null || distro_pkgs_to_install+="wget "
+  pip freeze | grep tox &>/dev/null || py_pkgs_to_install+="tox==3.3.0 "
```

Comment:
Can we use constraints here?

Reply:
We could. Are you thinking an in-repo set of constraints?

Reply:
I was thinking pointing upstream somewhere so we don't have another thing to maintain (within the file here, or our repo).

Reply:
Well, pointing to a web address might be nice, but it does introduce another opportunity for failure. We could easily use the dep_update job to prepare the constraints file and propose changes along with all other dependencies. It should be a small set. I don't see tox in https://github.com/openstack/requirements/blob/master/upper-constraints.txt anyway.

Reply:
Could work (dep_update), also funny that we don't have tox I don't see anything preventing us from adding it.

I'd like to see reuse of the requirements code for managing constraints if possible (that's one of the things I'm working on upstream).

Reply:
Perhaps it'd be best to create a new card to implement this then, and leave this as-is? It's quite a divergence from the focus of my current card. For the tox install, all I did was move it and set a pin here - previously there was no pin.

Reply:
As long as we are tracking it, sure.

Reply:
I'm suggesting that you register a card in RI given that it's something you'd like to see as an improvement and have the domain knowledge for it. :)

Reply:
I'd be happy to help out, of course.

Reply:
lol, I knew it'd come down to this :P

Reply:
For reference, the card is here: https://rpc-openstack.atlassian.net/browse/RI-475